src/plugins/layout/shadowdom.ts

```diff
@@ -365,8 +367,11 @@ export class ShadowDom {
     let newNodes = this.doc.getElementsByClassName(NewNodeClassName);
     for (let i = 0; i < newNodes.length; i++) {
       let shadowNode = newNodes[i] as IShadowDomNode;
+      let currentIndex = getNodeIndex(shadowNode);
       shadowNode.id = "" + nextIndex;
       shadowNode.node[NodeIndex] = nextIndex;
+      delete this.nodes[currentIndex];
```

Comment:
Ran some tests and looks like it is, indeed, almost twice faster. The only issue is that setting some property (or array index) to undefined, still counts as having that property in the object/array with value 'undefined'. So, if we do this:

var a = [];
a[5] = 5;
a[5] = undefined;

Then, a.length === 6. Don't think we're using nodes.length anywhere right now, but this could be a bit dangerous if someone later on decides to query the number of observed nodes through nodes.length.